node/node_test.go

```diff
@@ -35,15 +35,15 @@ func initializeAndStartLightNode(ctx context.Context, t *testing.T) *LightNode {
 
 // initializeAndStartNode initializes and starts a test node
 func initializeAndStartNode(ctx context.Context, t *testing.T, nodeType string) Node {
-	node := setupTestNode(ctx, t, nodeType)
+	node, _ := setupTestNode(ctx, t, nodeType)
 	require.False(t, node.IsRunning())
 	err := node.Start()
 	require.NoError(t, err)
 	require.True(t, node.IsRunning())
 	return node
 }
 
-func newTestNode(ctx context.Context, t *testing.T, nodeType string) (Node, error) {
+func newTestNode(ctx context.Context, t *testing.T, nodeType string) (Node, error, ed25519.PrivKey) {
```

Comment:
```suggestion
func newTestNode(ctx context.Context, t *testing.T, nodeType string) (Node, ed25519.PrivKey, error) {
```
we should keep the convention of ending with an error.

Reply:
fixed